### PR TITLE
[Merged by Bors] - feat: MulPosMono (WithZero α)

### PR DESCRIPTION
**Pull request by kbuzzard**
These instances make it possible to argue with multiplication and inequalities in `ℤₘ₀` .

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
Could you move that to the folder `Algebra.Order.GroupWithZero`? Maybe in the file `Algebra.Order.GroupWithZero.WithZero`?

File: Mathlib/Algebra/Order/GroupWithZero/WithZero.lean
Comment (YaelDillies) [inline Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:29]:
```suggestion
-/

assert_not_exists Ring
```

Comment (YaelDillies) [reply Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:26]: I only eat rings in my soup

(`Semiring` and `Ring` are defined in the same file)

**Review by YaelDillies — APPROVED** [review]
maintainer merge

File: Mathlib/Algebra/Order/GroupWithZero/WithZero.lean
Comment (faenuccio) [inline Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:36]:
```suggestion
        simpa only [mul_zero] using WithZero.zero_lt_coe _
```

File: Mathlib/Algebra/Order/GroupWithZero/WithZero.lean
Comment (faenuccio) [inline Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:28]: Is this really supposed to survive in mathlib once the file compiles?

File: Mathlib/Algebra/Order/GroupWithZero/WithZero.lean
Comment (faenuccio) [inline Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:36]:
```suggestion
        simpa only [mul_zero] using WithZero.zero_lt_coe _
```

File: Mathlib/Algebra/Order/GroupWithZero/WithZero.lean
Comment (faenuccio) [inline Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:47]:
```suggestion
        simpa only [mul_zero] using WithZero.zero_lt_coe _
```

File: Mathlib/Algebra/Order/GroupWithZero/WithZero.lean
Comment (riccardobrasca) [reply Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:28]: Yes, the point is to check that this file does not import `Ring`.

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

File: Mathlib/Algebra/Order/GroupWithZero/WithZero.lean
Comment (faenuccio) [reply Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:28]: Well, but I understand this while preparing the file. If it imports it, CI would give an error, no?

Comment (kbuzzard) [reply Mathlib/Algebra/Order/GroupWithZero/WithZero.lean:28]: My understanding is that if someone adds an import to some other file at some later time, and this happens to end up with this file importing Ring, then we discover this before the import addition is merged.

**Comment by kbuzzard** [comment]
Thanks all!

bors r+

**Pull request closed without merging**
